### PR TITLE
fix for glitchy grabs and failure to take simulation ownership

### PR DESCRIPTION
**Pull request by AndrewMeadows**
This appears to fix the glitchy grabs and also helps your interface become the simulation owner for objects that it is manipulating.

### CODE REVIEW

File: libraries/entities/src/EntityItem.cpp [hunk 1]
```diff
@@ -337,299 +328,308 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
...
+    READ_ENTITY_PROPERTY(PROP_USER_DATA, QString, setUserData);
+
+    if (args.bitstreamVersion >= VERSION_ENTITIES_HAVE_ACCELERATION) {
+        // we always accept the server's notion of simulatorID, so we fake overwriteLocalData as true
+        // before we try to READ_ENTITY_PROPERTY it 
+        bool temp = overwriteLocalData;
+        overwriteLocalData = true;
+        READ_ENTITY_PROPERTY(PROP_SIMULATOR_ID, QUuid, updateSimulatorID);
+        overwriteLocalData = temp;
+    }
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: Most of the diffs here are just indentation changes.  One of the real fixes is above here where we fake the value of overwriteLocalData before we read the SimulatorID property.  This property is always authoritative from the EntityServer so we should never ignore it.

File: libraries/entities/src/EntityItem.cpp [hunk 2]
```diff
@@ -337,299 +328,308 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
...
         _position = savePosition;
         _rotation = saveRotation;
-        // _velocity = saveVelocity;
-        // _angularVelocity = saveAngularVelocity;
-        // _gravity = saveGravity;
-        // _acceleration = saveAcceleration;
+        _velocity = saveVelocity;
+        _angularVelocity = saveAngularVelocity;
+        _dirtyFlags &= ~(EntityItem::DIRTY_TRANSFORM | EntityItem::DIRTY_VELOCITIES);
     }
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: An important fix is to enforce our own transform AND velocities when we own the simulation.  This fixes a glitch when moving objects accidentally get an update from the EntityServer that gets by the "lastEdited is too old" checks.  We don't want to mitigate fighting over an object's velocity between two interfaces by ignoring our own notion of velocity when we own the simualtion -- that path leads toward insanity.  Instead we should register Actions and each simulator will have to compute all relevant Actions themselves in order to predict an object's motion (future work).

File: libraries/entities/src/EntityItem.cpp [hunk 3]
```diff
@@ -948,40 +948,25 @@ bool EntityItem::setProperties(const EntityItemProperties& properties) {
...
-        timestamp = properties.getLastEdited();
-        if (timestamp > now) {
-            timestamp = now;
-        } else if (timestamp < _created) {
-            timestamp = _created;
-        }
-        _lastEdited = timestamp;
-    }
-
     return somethingChanged;
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: This stuff not needed anymore.  We set _created above when something changed.

File: libraries/physics/src/EntityMotionState.cpp [hunk 4]
```diff
@@ -401,13 +400,12 @@ void EntityMotionState::sendUpdate(OctreeEditPacketSender* packetSender, const Q
     properties.setAcceleration(_serverAcceleration);
     properties.setAngularVelocity(_serverAngularVelocity);
 
-    // we only update lastEdited when we're sending new physics data 
-    quint64 lastSimulated = _entity->getLastSimulated();
-    _entity->setLastEdited(lastSimulated);
-    properties.setLastEdited(lastSimulated);
+    // set the LastEdited of the properties but NOT the entity itself
+    quint64 now = usecTimestampNow();
+    properties.setLastEdited(now);
```
Comment (AndrewMeadows) [inline libraries/physics/src/EntityMotionState.cpp]: Another important fix is to NOT set the entity's LastEdited time when updating its positions from simulation.  Instead we only set the outgoing properties which are being sent to the EntityServer.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0a0eba8277b7d098cfa764a4b8e80d0afecb0b29 available at: https://gist.github.com/7edfd5ba22db76ce20d6<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4983.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4983.dmg

**Pull request merged**